spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    [Verb("queryNewColumn", HelpText = "Query data from a newly added column in the sample Spanner database table.")]
+    class QueryNewColumnOptions
+    {
+        [Value(0, HelpText = "The project ID of the project to use when managing Spanner resources.", Required = true)]
+        public string projectId { get; set; }
+        [Value(1, HelpText = "The ID of the instance where the sample database resides.", Required = true)]
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    [Verb("queryDataWithTransaction", HelpText = "Query the sample Spanner database table using a transaction.")]
+    class QueryDataWithTransactionOptions
+    {
+        [Value(0, HelpText = "The project ID of the project to use when managing Spanner resources.", Required = true)]
+        public string projectId { get; set; }
+        [Value(1, HelpText = "The ID of the instance where the sample database resides.", Required = true)]
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    [Verb("readWriteWithTransaction", HelpText = "Update data in the sample Spanner database table using a read-write transaction.")]
+    class ReadWriteWithTransactionOptions
+    {
+        [Value(0, HelpText = "The project ID of the project to use when managing Spanner resources.", Required = true)]
+        public string projectId { get; set; }
+        [Value(1, HelpText = "The ID of the instance where the sample database resides.", Required = true)]
+        public string instanceId { get; set; }
+        [Value(2, HelpText = "The ID of the database where the sample database resides.", Required = true)]
+        public string databaseId { get; set; }
+    }
+
+    public class Program
+    {
+        public static async Task<object> CreateSampleDatabaseAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START create_database]
+            // Create client.
+            DatabaseAdminClient databaseAdminClient =
+                await DatabaseAdminClient.CreateAsync();
+            // Initialize request argument(s).
+            InstanceName parent = new InstanceName(projectId, instanceId);
+            string createStatement = $"CREATE DATABASE `{databaseId}`";
+            // Make the request.
+            var createDatabaseRequest =
+                await databaseAdminClient.CreateDatabaseAsync(parent,

```

Comment:
this is using the vkit api which is not as idiomatic as the ado.net wrappers.  I cannot see why the vkit api doesnt let you simply pass the db name other than its a straight wrapper generated on the proto.  We could enhance it.

However, the ado.net version lets you pass any ddl statement -- it isn't specific to createdatabase, etc.

I would prefer all the samples to use the ado.net version as it will complicate matters if developers start taking direct dependencies on our vkit layer, which we have not vetted as much.